### ISSUE TITLE
Kiss-ICP with a mem limit is permanently GC-ing because _something_ is hogging memory

### ISSUE THREAD

**Issue by teh-cmc** [rerun-io/rerun#5685]
Via @02alexander:
![image](https://github.com/rerun-io/rerun/assets/2910679/a2b90e5b-61bc-4ad7-9311-ebc3dc31fe38)

which basically means that we're GC-ing like crazy in hopes of reducing memory, except the thing that's actually hogging memory is not part of the GC path.

Hence the question is: what is the something in question? :eye:

**Comment by teh-cmc** [comment]
This feels _very much_ like:
- https://github.com/rerun-io/rerun/issues/4185

**Comment by teh-cmc** [comment]
TL;DR: Fixed by removing instance keys:
- https://github.com/rerun-io/rerun/issues/5303

---

So, here's what happening.

The Kiss-ICP pipeline logs a new, bigger point-cloud as its `/world/map` every frame.
(I assume it does so because it either can't or doesn't want to rely on visible history for whatever reason.)

`/world/map` number of instances over time:
![image](https://github.com/rerun-io/rerun/assets/2910679/f0f6caa4-3c2f-460a-89a8-4a44739473f9)


Because that number is different every frame, we need to generate a new instance key batch matching that number of instances, every frame.
And because that number is quite large, and only getting larger and larger, that's a lot of data.
And because autogenerated keys are cached and unaffected by GC, that's a lot of data hangs around permanently!

That's what you see on the left side of this memgraph:
![image](https://github.com/rerun-io/rerun/assets/2910679/f0d0230a-6c74-4755-a0a1-9e7968b1efcc)


Now, because that data hangs around indefinitely and is by itself larger than the GC threshold, the GC will now trigger every frame:
![image](https://github.com/rerun-io/rerun/assets/2910679/3755e1ec-67f2-42dd-8427-60d1dac11fdd)


Because the GC triggers every frame, the whole viewer slows downs, which means ingestion slows down, which mean raw arrow data starts to accumulate in the different layers of buffers/channels/etc.
That's why the memory permanently grows at a very alarming rate, even though the GC is running like crazy.

That's what you see on the other side of this memgraph:
![image](https://github.com/rerun-io/rerun/assets/2910679/f0d0230a-6c74-4755-a0a1-9e7968b1efcc)

---

So, one solution to this would be to modify Kiss-ICP to log its own instance keys, that way they would get GC'd as needed.
But that's not possible since we don't expose instance keys to the SDK anymore.

Another solution would be to make instance-key generation smarter by sharing their backing buffers etc.
But that's tricky considering we have a lot of address-based performance hacks around instance keys... and we're trying to get rid of instance keys to begin with!
- https://github.com/rerun-io/rerun/issues/1777

So the last is simply to wait for 0.16 to land, which should get rid of (autogenerated) instance keys entirely:
- https://github.com/rerun-io/rerun/issues/5303
